### DISCUSSION TITLE
Oracle - exec_driver_sql gives ORA-01036: illegal variable name/number when running single tuple with "?" in bind variables

### DISCUSSION THREAD

**Discussion by Maksymilian-Plywaczyk** [sqlalchemy/sqlalchemy#10642] [Usage Questions] [answered]
Hello :)

While inserting values into Oracle database, I found little trouble with bind variables. In SQLAlchemy documentation for single tuple it looks like this:
`conn.exec_driver_sql(
    "INSERT INTO table (id, value) VALUES (?, ?)",
    (1, 'v1')
)`

But in my case that gives me an error:  **ORA-01036: illegal variable name/number**.
What works for me is to change "?" for ":0" like this one:
`conn.exec_driver_sql(
    "INSERT INTO table (id, value) VALUES (:0, :1)",
    (1, 'v1')
)`

I'm not sure in this case is my fault (maybe I didn't look at something important), but I want to check that one. If not I would recommend changing that in your documentation :)

**Comment by CaselIT** [comment, marked as answer]
hi,

As indicated in the the documentation of the exec_driver_sql https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Connection.exec_driver_sql.params.statement bound parameter must follow the convention used by the dbapi, in the case the oracle one.

The example in the documentation uses sqlite that uses question marks, so it's correct as is.

if you want sqlalchemy to do that for you you should use `execute` and `test` using as parameters the named style `:name`, `:other_name` etc

**Reply by Maksymilian-Plywaczyk** [reply]
Thank you for your answer, and sorry for my fault that I didn't check all information in your documentation.
